### ISSUE TITLE
Changing .perltidyrc requires reload to take effect

### ISSUE THREAD

**Issue by FractalBoy** [FractalBoy/perl-language-server#51]
For some reason, when .perltidyrc is changed, it requires a reload for the change to take effect.

It's unclear if this is a bug in PLS or in Perl::Tidy. If it is a bug in Perl::Tidy, we might need to start a new process to run perltidy so that the config is not retained in memory.

**Comment by FractalBoy** [comment]
It does look like Perl::Tidy caches the result of processing command line arguments if the arguments are the same as a previous execution.

So we need to add some kind of argument to break the cache.

```
sub process_command_line {
 
    my @q = @_;
    my (
        $perltidyrc_stream,  $is_Windows, $Windows_type,
        $rpending_complaint, $dump_options_type
    ) = @q;
 
    my $use_cache = !defined($perltidyrc_stream) && !$dump_options_type;
    if ($use_cache) {
        my $cache_key = join( chr(28), @ARGV );
        if ( my $result = $process_command_line_cache{$cache_key} ) {
            my ( $argv, @retvals ) = @{$result};
            @ARGV = @{$argv};
            return @retvals;
        }
        else {
            my @retvals = _process_command_line(@q);
            $process_command_line_cache{$cache_key} = [ \@ARGV, @retvals ]
              if $retvals[0]->{'memoize'};
            return @retvals;
        }
    }
    else {
        return _process_command_line(@q);
    }
}
```

**Comment by FractalBoy** [comment]
Looks like you can bust the cache by always providing a perltidyrc file, even if it is the default.

**Comment by FractalBoy** [comment]
My guess is that we stopped getting the perltidyrc config when we moved it to an IO::Async::Function. Need to pass that as a parameter instead of getting it from the global variable.

**Comment by FractalBoy** [comment]
Fixed in 3e6e6ac97ffe38105908e285c665b524957160ed
